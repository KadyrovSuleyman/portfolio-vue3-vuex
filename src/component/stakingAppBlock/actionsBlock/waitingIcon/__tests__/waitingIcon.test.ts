import { mount, VueWrapper } from '@vue/test-utils';
import WaitingIcon from '../waitingIcon.vue';

let wrapper: VueWrapper<any>;
afterEach(() => {
  wrapper.unmount();
});

it('waitingIcon renders', () => {
  wrapper = mount(WaitingIcon);

  expect(wrapper.find('.waitingIcon').classes()).toEqual(['waitingIcon']);
  expect(wrapper.find('.fill').exists()).toBeTruthy();
  expect(wrapper.find('.active').exists()).toBeTruthy();
});

it('watchs props changes', async () => {
  wrapper = mount(WaitingIcon);
  expect(wrapper.find('.waitingIcon').classes()).toEqual(['waitingIcon']);

  await wrapper.setProps({
    mods: {
      selected: true,
    },
  });
  expect(wrapper.find('.waitingIcon').classes()).toEqual(['waitingIcon', 'waitingIcon__selected']);

  await wrapper.setProps({
    mods: {
      selected: false,
    },
  });
  expect(wrapper.find('.waitingIcon').classes()).toEqual(['waitingIcon']);
});
