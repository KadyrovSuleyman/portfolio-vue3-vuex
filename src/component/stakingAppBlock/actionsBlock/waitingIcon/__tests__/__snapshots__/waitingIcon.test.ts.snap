// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`waitingIcon renders 1`] = `
<div class="waitingIcon">
  <div class="fill"></div>
  <div class="active"></div>
</div>
`;
